sql/src/test/java/org/apache/druid/sql/calcite/CalciteInsertDmlTest.java

```diff
@@ -1638,6 +1639,9 @@ public void testErrorWithUnableToConstructColumnSignatureWithExtern()
                                 "general"
                             )
                                 .expectMessageContains(
+                                    JvmUtils.majorVersion() >= 17
+                                    ? "Cannot construct instance of `org.apache.druid.segment.column.ColumnSignature`, problem: Cannot invoke \"String.length()\" because \"s\" is null"
+                                    :
```

Comment:
Can we change the source of the exception to do something that doesn't cause `NPE`s?  There's no way that an end user woudl actually understand what `s` is and why it needs `length` called on it.  Come to think of it, there's probably no reason to believe that they would even understand what an `org.apache.druid.segment.column.ColumnSignature` is...